File: CS50/pset6/caesar/caesar.py
import cs50
import sys

key = int(sys.argv[1])

encrypt = cs50.get_string("What would you like to encrypt?\n")

encrypted = ""

for letter in encrypt:
    if letter.isalpha():

        if letter.isupper():
            charact = (ord(letter) - 65 +  key) % 26
            encrypted_char = chr(charact+65)
            encrypted += (encrypted_char)

        elif letter.islower():
            charact = (ord(letter) - 97 +  key) % 26
            encrypted_char = chr(charact+97)
            encrypted += (encrypted_char)
    else:

        encrypted += (letter)


print(encrypted)

File: CS50/pset2/caesar/caesar.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>

int main(int argc, string argv[])
{

    int n = argc;
    printf("%i", argc);
    if (n != 2)
    {
        printf("\nError 1, you must supply one number for the key");
        return 1;
    }
    else
    {
        int k = atoi(argv[1]);
        string p = get_string("Please enter what you want to encrypt: ");
        int plain_len = strlen(p);
        char c[plain_len];

        // char new_char;

        for (int i = 0; i < plain_len; i++)
        {
            if (isalpha(p[i]))
            {

                if (isupper(p[i]))
                {
                    //converting to an alphabetical index (a = 0, b = 1, etc..)
                    char charact = (p[i] - 65 + k) % 26;
                    c[i] = charact + 65;
                }
                if (islower(p[i]))
                {
                    char charact = (p[i] - 97 + k) % 26;
                    c[i] = charact + 97;

                }




            }
            else
            {
                c[i] = p[i];
            }
            printf("%c", c[i]);





        }

    }
}